core/Test/Tasty/Core.hs

```diff
@@ -361,22 +377,31 @@ foldTestTree
   -> TestTree
      -- ^ the tree to fold
   -> b
-foldTestTree (TreeFold fTest fGroup fResource fAfter) opts0 tree0 =
-  go mempty opts0 tree0
+foldTestTree (TreeFold fTest fGroup fAfter) opts0 tree0 =
+  snd (go mempty opts0 mempty tree0)
```

Comment:
It gets a bit convoluted. Can we decompose this function into two passes: one to apply pattern `TestPattern -> TestTree -> TestTree`, and another to actually fold?

Reply:
I'm not sure what this would look like. Could you spell it out a bit more? If it helps, I've added inline documentation.

Reply:
See https://github.com/UnkindPartition/tasty/pull/363. I imagine you should be able to modify `filterByPattern` only.